hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -478,7 +524,8 @@ else if ( fetchDepth > maximumFetchDepth ) {
 				);
 				fetches.add( fetch );
 
-				if ( fetchable instanceof PluralAttributeMapping && fetchTiming == FetchTiming.IMMEDIATE ) {
+				if ( fetchTiming == FetchTiming.IMMEDIATE && joined && fetchable instanceof PluralAttributeMapping ) {
+					applyFiltering( querySpec, loadQueryInfluencers, (PluralAttributeMapping) fetchable );
```

Comment:
The first bug flushed out. We only wanna apply `filtering` and `ordering` for `join` type, not for `select` type.